tracer/contrib/database/sqltraced/utils.go

```diff
@@ -0,0 +1,136 @@
+package sqltraced
+
+import (
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
+	"golang.org/x/net/context"
+)
+
+// Return all information passed through the DSN:
+func parseDSN(driver interface{}, dsn string) (meta map[string]string, err error) {
+	switch driver.(type) {
+	case *pq.Driver:
+		meta, err = parsePostgresDSN(dsn)
+	case *mysql.MySQLDriver:
+		meta, err = parseMySQLDSN(dsn)
+	}
+	meta = normalize(meta)
+	return
+}
+
+func normalize(meta map[string]string) map[string]string {
+	// Delete the entry "password" for security reasons
+	delete(meta, "password")
+
+	m := make(map[string]string)
+	for k, v := range meta {
+		m[normalizeKey(k)] = v
+	}
+	return m
+}
+
+func normalizeKey(k string) string {
+	switch k {
+	case "user":
+		return "db." + k
```

Comment:
you can skip the concatenation and just return `db.user` and it will be more efficient.